plenum/test/node/test_quota_control.py

```diff
@@ -0,0 +1,50 @@
+from plenum.server.quota_control import QuotaControl
+
+MAX_NODE_MESSAGE_COUNT_LIMIT = 100
+MAX_NODE_MESSAGE_SIZE_LIMIT = 1024 * 1024
+MAX_CLIENT_MESSAGE_COUNT_LIMIT = 20
+MAX_CLIENT_MESSAGE_SIZE_LIMIT = 24 * 1024
+
+
+def create_quota_control() -> QuotaControl:
+    return QuotaControl(max_node_message_count_limit=MAX_NODE_MESSAGE_COUNT_LIMIT,
+                        max_node_message_size_limit=MAX_NODE_MESSAGE_SIZE_LIMIT,
+                        max_client_message_count_limit=MAX_CLIENT_MESSAGE_COUNT_LIMIT,
+                        max_client_message_size_limit=MAX_CLIENT_MESSAGE_SIZE_LIMIT)
+
+
+def test_quota_control_gives_maximum_quotas_initially():
+    qc = create_quota_control()
+    assert qc.node_message_count_limit == MAX_NODE_MESSAGE_COUNT_LIMIT
+    assert qc.node_message_size_limit == MAX_NODE_MESSAGE_SIZE_LIMIT
+    assert qc.client_message_count_limit == MAX_CLIENT_MESSAGE_COUNT_LIMIT
+    assert qc.client_message_size_limit == MAX_CLIENT_MESSAGE_SIZE_LIMIT
+
+
+def test_quota_control_gives_maximum_quotas_when_none_are_reached():
+    qc = create_quota_control()
+    qc.received_node_messages(MAX_NODE_MESSAGE_COUNT_LIMIT // 2, MAX_NODE_MESSAGE_SIZE_LIMIT // 2)
```

Comment:
I would prefer to have here ` qc.received_node_messages(MAX_NODE_MESSAGE_COUNT_LIMIT - 1 MAX_NODE_MESSAGE_SIZE_LIMIT - 1)`